mmv1/third_party/terraform/tests/resource_network_connectivity_service_connection_policies_test.go

```diff
@@ -0,0 +1,134 @@
+	t.Parallel()
+
+	networkName := fmt.Sprintf("tf-test-network-%s", RandString(t, 10))
+	networkProducerName := fmt.Sprintf("tf-test-network-%s", RandString(t, 10))
+	subnetworkConsumerName := fmt.Sprintf("tf-test-subnet-consumer-%s", RandString(t, 10))
+	subnetworkProducerName := fmt.Sprintf("tf-test-subnet-producer-%s", RandString(t, 10))
+	serviceConnectionPolicyName := fmt.Sprintf("tf-test-service-connection-policy-%s", RandString(t, 10))
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckNetworkConnectivityServiceConnectionPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetworkConnectivityServiceConnectionPolicy_basic(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName),
+			},
+			{
+				ResourceName:      "google_network_connectivity_service_connection_policy.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccNetworkConnectivityServiceConnectionPolicy_update(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName),
+			},
+			{
+				ResourceName:      "google_network_connectivity_service_connection_policy.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccNetworkConnectivityServiceConnectionPolicy_basic(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName),
+			},
+			{
+				ResourceName:      "google_network_connectivity_service_connection_policy.default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNetworkConnectivityServiceConnectionPolicy_basic(networkName, subnetworkConsumerName, networkProducerName, subnetworkProducerName, serviceConnectionPolicyName string) string {
+	return fmt.Sprintf(`
+  resource "google_compute_network" "consumer_net" {
+    name                    = "%s"
+    auto_create_subnetworks = false
+  }
+  
+  resource "google_compute_subnetwork" "consumer_subnet" {
+    name          = "%s"
+    ip_cidr_range = "10.0.0.0/16"
+    region        = "us-central1"
+    network       = google_compute_network.consumer_net.id
+  }
+  
+  resource "google_compute_network" "producer_net" {
+    name                    = "%s"
+    auto_create_subnetworks = false
+  }
+  
+  resource "google_compute_subnetwork" "producer_subnet" {
+    name          = "%s"
+    ip_cidr_range = "10.0.0.0/16"
+    region        = "us-central1"
+    network       = google_compute_network.producer_net.id
+  }
+  
+  resource "google_network_connectivity_service_connection_policy" "default" {
+    name = "%s"
+    location = "us-central1"
+    description = "my basic service connection policy"
```

Comment:
Nit: It usually covers a few more edge cases to have a test like this specify the minimum set of fields (so you could probably remove `description`, maybe others). It will also help validate some of my early questions regarding which fields are actually required.

Reply:
Do you believe this test is not necessary to exist? I'm just adding labels to test the  update.

Reply:
We actually have a requirement to test updating any fields that are updatable, so I would definitely keep this, but my suggestion is to simply remove the `description` on this line so that the test hits a few more edges of the implementation. You may also be able to remove other fields here that are not marked required, but I'm not sure.